File: Axiom/Internal/Ast/BlockStatement.cs
using System.Collections.Generic;

namespace Axiom.Internal.Ast
{
    internal class BlockStatement : Statement
    {
        public BlockStatement(IList<Statement> statements)
        {
            Statements = statements;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public IList<Statement> Statements { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/BlockStatement.java
package com.moybl.axiom.ast;

import java.util.List;

public class BlockStatement implements Statement {

	private List<Statement> statements;

	public BlockStatement(List<Statement> statements) {
		this.statements = statements;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public List<Statement> getStatements() {
		return statements;
	}

}
